esbuilder/lit-connect-modal/src/modal.js

```diff
@@ -0,0 +1,163 @@
+import MicroModal from 'micromodal';
+import css from './modal.css';
+import {
+  rawListOfWalletsArray,
+  metaMaskSingle,
+  coinbaseSingle,
+} from './helpers/walletList.js';
+import providerMethods from './helpers/providerMethods.js';
+
+export default class LitConnectModal {
+  constructor({ providerOptions }) {
+    this.dialog = MicroModal;
+    this.closeAction = undefined;
+    this.parent = document.body;
+    this.filteredListOfWalletsArray = [];
+    this.providerOptions = providerOptions;
+    this._filterListOfWallets();
+    this._instantiateLitConnectModal();
+
+    // inject css
+    var style = document.createElement('style');
+    style.innerHTML = css;
+    document.head.appendChild(style);
+  }
+
+  getWalletProvider() {
+    const currentProvider = localStorage.getItem('lit-web3-provider');
+
+    this.dialog.show('lit-connect-modal');
+    return new Promise((resolve, reject) => {
+      // if there is a current provider, resolve with it
+      if (!!currentProvider) {
+        const foundProvider = this.filteredListOfWalletsArray.find(
+          (w) => w.id === currentProvider
+        );
+        resolve(foundProvider.provider);
+        this._destroy();
+        return;
+      }
+
+      // otherwise, show the list of providers
+      this.filteredListOfWalletsArray.forEach((w) => {
+        let walletEntry = document.getElementById(w.id);
+        walletEntry.addEventListener('click', () => {
+          localStorage.setItem('lit-web3-provider', w.id);
+          resolve(w.provider);
+          this._destroy();
+          return;
+        });
+      });
+
+      this.closeAction.addEventListener('click', () => {
+        resolve(false);
+        this._destroy();
+        return;
+      });
+    });
+  }
+
+  _filterListOfWallets() {
+    const filteredListOfWalletsArray = [];
+
+    // -- this would only work if user installed multiple wallet extensions
+    // eg. when "ethereum.providers" is defined
+    rawListOfWalletsArray.forEach((w) => {
+      if (!!w['checkIfPresent'] && w['checkIfPresent']() === true) {
```

Comment:
is there a reason we need to key into the object instead of using standard '.' notation for property access here?

Reply:
not sure, probably this still allow you to check if when `checkIfPresent` is not present, just another way to do `?.checkIfPresent`

Reply:
We might want to change to '?.' property notation to stay consistent.